Jenkinsfile

```diff
@@ -465,17 +473,90 @@ EXIT /b 1
+                      remoteDirectory: ".tmp/${env.GIT_BRANCH}",
+                      removePrefix: "install/linux/doc",
+                      sourceFiles: 'install/linux/doc/**')
+                  ]
+                )
+              ]
+            )
+          }
+        }
+
+        stage('upload-artifacts') {
+          when {
+            allOf {
+              not {
+                changeRequest()
+              }
+              anyOf {
+                buildingTag()
+                anyOf {
+                  branch 'master'
+                  branch 'maintenance/**'
+                }
+              }
+              expression { return params.MINGW32 }
+              expression { return params.UPLOAD_BUILD_OPENMODELICA }
+            }
+            beforeAgent true
+          }
+          agent {
+            label 'linux'
+          }
+          steps {
+            unstash name: 'amd64-zip'         // includes: "OMSimulator-linux-amd64-*.tar.gz"
+            unstash name: 'arm32-zip'         // includes: "OMSimulator-linux-arm32-*.tar.gz"
+            unstash name: 'i389-zip'          // includes: "OMSimulator-linux-i386-*.tar.gz"
+            unstash name: 'mingw32-zip'       // includes: "OMSimulator-mingw32*.zip"
+            unstash name: 'mingw64-zip'       // includes: "OMSimulator-mingw64*.zip"
+            unstash name: 'win64-zip'         // includes: "OMSimulator-win64*.zip"
+            unstash name: 'osx-zip'           // includes: "OMSimulator-osx*.zip"
+
+            sh "ls *.zip *.tar.gz"
+
+            sshPublisher (
+              publishers: [
+                sshPublisherDesc(
+                  configName: 'OMSimulator',
+                  transfers: [
+                    sshTransfer(
+                      remoteDirectory: "linux-i368/",
+                      sourceFiles: 'OMSimulator-linux-i386-*.tar.gz'),
+                    sshTransfer(
+                      remoteDirectory: "linux-arm32/",
+                      sourceFiles: 'OMSimulator-linux-arm32-*.tar.gz'),
+                    sshTransfer(
+                      remoteDirectory: "linux-amd64/",
+                      sourceFiles: 'OMSimulator-linux-amd64-*.tar.gz'),
+                    sshTransfer(
+                      remoteDirectory: "win-mingw32/",
+                      sourceFiles: 'OMSimulator-mingw32*.zip'),
+                    sshTransfer(
+                      remoteDirectory: "win-mingw64/",
+                      sourceFiles: 'OMSimulator-mingw64*.zip'),
+                    sshTransfer(
+                      remoteDirectory: "osx/",
+                      sourceFiles: 'OMSimulator-osx*.zip'),
+                    sshTransfer(
+                      remoteDirectory: "win-msvc64/",
+                      sourceFiles: 'OMSimulator-win64*.zip')
```

Comment:
Do we need to create the directories on build.openmodelica.org manually?

Reply:
Not if you create the directory locally and transfer the `sourceFiles: 'linux-i386/OMSimulator-linux-i386-*.tar.gz'` I think. Also note that it's not `i368`

Reply:
The ssh transfer creates all directories. At least it did when I uploaded to the wrong location.